src/c++/perf_analyzer/docs/examples/profile.py

```diff
@@ -0,0 +1,154 @@
+        "--measurement-mode=count_windows "
+        "--measurement-request-count=10 "
+        "--stability-percentage=999"
+    )
+    ret = subprocess.run(args=[command], shell=True)
+    ret.check_returncode()
+
+
+def generate_input_data(args, filename):
+    request_parameters = f"""
+    {{
+        "max_tokens": {args.max_tokens},
+        "ignore_eos": {"true" if args.ignore_eos else "false"}
+    }}
+    """
+    input_data = {"data": [{"STREAM": [True]}]}
+    input_data["data"][0]["SAMPLING_PARAMETERS"] = [request_parameters]
+
+    prompt = ["hi"] * prompt_size  # Generate dummy prompt
+    input_data["data"][0]["PROMPT"] = [" ".join(prompt)]
+    with open(filename, "w") as f:
+        json.dump(input_data, f)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-m",
+        "--model",
+        type=str,
+        default="vllm",
+        choices=["vllm"],
+        help="The name of the model to profile.",
+    )
+    parser.add_argument(
+        "--prompt-size-range",
+        type=int,
+        nargs=3,
+        metavar=("START", "END", "STEP"),
+        default=[10, 10, 1],
+        help="The range of prompt sizes '<[START, END], STEP>' where END is inclusive.",
+    )
+    parser.add_argument(
+        "--max-tokens",
+        type=int,
+        default=256,
+        help="The maximum number of tokens to generate.",
+    )
+    parser.add_argument(
+        "--ignore-eos",
+        action="store_true",
+        help="Whether to ignore end-of-sequence token.",
+    )
+    parser.add_argument(
+        "--input-data",
+        type=str,
+        default=None,
+        help="The input data file to be used for inference request.",
+    )
+    args = parser.parse_args()
+
+    results = []
+
+    if args.input_data:
+        print(f"Using input data file '{args.input_data}' for inference request.\n")
+        with open(args.input_data) as f:
+            input_data = json.load(f)
+            prompt_size = len(input_data["data"][0]["PROMPT"][0].split())
+            args.prompt_size_range = [prompt_size, prompt_size, 1]
+
+    start, end, step = args.prompt_size_range
+    for prompt_size in range(start, end + 1, step):
+        if not args.input_data:
+            generate_input_data(args, TEMP_INPUT_FILE)
+
+        profile(args, args.input_data if args.input_data else TEMP_INPUT_FILE)
+        avg_first_token_latency, avg_token_to_token_latency = calculate_avg_latencies()
+        results.append(
+            (prompt_size, avg_first_token_latency, avg_token_to_token_latency)
+        )
+
+    print("\n[ Benchmark Summary ]")
+    for prompt_size, avg_first_token_latency, avg_token_to_token_latency in results:
+        print(
+            f"  Prompt size: {prompt_size}, "
+            f"Average first-token latency: {avg_first_token_latency:.4f} sec, "
+            f"Average token-token latency: {avg_token_to_token_latency:.4f} sec"
```

Comment:
Should average token-to-token latency be omitted when max tokens is 1? i.e. there shouldn't be any token-to-token latency.

There might be a case where technically two responses are returned when max tokens is set to 1 but that's because the second token is a null response with the final response flag set to true. I don't think the latency between that response and the one true response before it is useful to the user.

Reply:
I would expect that avg and first t-t-t latency would be the same. I wouldn't omit this as some users might get confused and think the tool is broken when this doesn't get printed out.

Reply:
Yes

Reply:
I would expect them to be different. First token latency is between the request and the first response from the model and token to token is between the two consecutive responses.

Reply:
When max_tokens = 1, it's not undefined but rather meaningless information for the user since the last response is just a null response from the backend (and not the response for generating tokens). 

@matthewkotila I excluded the last token from the latency calculation. User will see avg token-token latency output when it's max_tokens > 1.